Update README with test setup instructions

aniketd opened the pull request:
cc. @lehins

aniketd committed:
Update README with test setup instructions

teodanciu reviewed (approved):
Looks good to me

JaredCorduan reviewed (approved):
thank you @aniketd ! :raised_hands:

JaredCorduan commented on README.md:
```diff
@@ -145,6 +145,10 @@ You can update the `sources.json` file using niv:
 
+# Testing
+
+Before running tests with `cabal test`, the `CARDANO_MAINNET_MIRROR` environment variable should be set to the path to [mainnet epochs](https://github.com/input-output-hk/cardano-mainnet-mirror/tree/master/epochs) on your local storage.
```
```suggestion
Before running tests with `cabal test`, the `CARDANO_MAINNET_MIRROR` environment variable should be set to the path to [mainnet epochs](https://github.com/input-output-hk/cardano-mainnet-mirror/tree/master/epochs) on your local storage.
Note that this is only needed for the Byron tests.
```

aniketd committed:
Update README.md

Add note about CARDANO_MAINNET_MIRROR
environment variable being required only by byron tests

Co-authored-by: Jared Corduan <jaredcorduan@gmail.com>
